reps/2022-10-11-serve-java-http-ingress.md

```diff
@@ -0,0 +1,167 @@
+## Summary
+### General Motivation
+We want to call various methods of java deployment through http proxy. Like the fastapi ingress in python serve.
+### Should this change be within `ray` or outside?
+main `ray` project. Changes are made to Ray Serve module.
+
+## Stewardship
+### Required Reviewers
+@simon-mo
+### Shepherd of the Proposal (should be a senior committer)
+@simon-mo
+
+## Design and Architecture
+### user case
+1. define the model with JAX-RS
+```java
+@Path("user")
+public class UserRestService {
+    @GET
+    @Path("helloWorld")
+    @Consumes({"application/json"})
+    @Produces({"application/json"})
+    public String helloWorld(String name) {
+        return "hello world, " + name;
+    }
+
+    @GET
+    @Path("paramPathTest/{name}")
+    @Consumes({"application/json"})
+    @Produces({"application/json"})
+    public String paramPathTest(@PathParam("name")String name) {
+        return "paramPathTest, " + name;
+    }
+}
+```
+2. create deployment
+```java
+Deployment deployment =
+        Serve.deployment()
+            .setName("deploymentName")
+            .setDeploymentDef(UserRestService.class.getName())
+            .ingress("jax-rs")
+            .setNumReplicas(1)
+            .create();
+deployment.deploy(true);
```

Comment:
nit: Add comment what does `true` mean here.

Reply:
It is similar to `_blocking` when deploying deployment with python serve.  This parameter already exists on the master branch of the ray project.
![image](https://user-images.githubusercontent.com/11265783/199513564-093c7016-66b8-4d7a-98ad-9759476dd88f.png)
![image](https://user-images.githubusercontent.com/11265783/199513752-39b7113a-cf0c-4bf5-8d78-573bdc754f91.png)